File: web/src/test/java/no/nav/sbl/dialogarena/rest/feil/ThrowableMapperTest.java
package no.nav.sbl.dialogarena.rest.feil;

import no.nav.sbl.sosialhjelp.SamtidigOppdateringException;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import javax.ws.rs.core.Response;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(MockitoJUnitRunner.class)
public class ThrowableMapperTest {

    final private ThrowableMapper mapper = new ThrowableMapper();

    @Test
    public void skalGi409ConflictVedSamtidigOppdateringException() {
        Response response = mapper.toResponse(new SamtidigOppdateringException("Mulig versjonskonflikt..."));
        assertThat(response.getStatus()).isEqualTo(409);
    }
}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/app/feil/ThrowableMapperTest.kt
package no.nav.sosialhjelp.soknad.app.feil

import no.nav.sosialhjelp.soknad.app.exceptions.SamtidigOppdateringException
import no.nav.sosialhjelp.soknad.app.rest.feil.ThrowableMapper
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class ThrowableMapperTest {

    private val mapper = ThrowableMapper("loginserviceUrl")

    @Test
    fun skalGi409ConflictVedSamtidigOppdateringException() {
        val response = mapper.toResponse(SamtidigOppdateringException("Mulig versjonskonflikt..."))
        assertThat(response.status).isEqualTo(409)
    }
}
